src/base/types/derivedmodels/SMomentModel.jl

```diff
@@ -0,0 +1,272 @@
+"""
+    mutable struct SMomentModel <: MetabolicModel
+
+Construct an enzyme capacity constrained model see `Bekiaris, Pavlos Stephanos,
+and Steffen Klamt. "Automatic construction of metabolic models with enzyme
+constraints." BMC bioinformatics, 2020.` for implementation details.
+
+Note, `"§"` is reserved for internal use as a delimiter, no reaction id should
+contain that character. Also note, SMOMENT assumes that each reaction only has a
+single enzyme (one GRR) associated with it. It is required that a model be
+modified to ensure that this condition is met. For ease-of-use,
+[`remove_slow_isozymes!`](@ref) is supplied to effect this. Currently only
+`modifications` that change attributes of the `optimizer` are supported.
+
+# Fields
+```
+reaction_ids::Vector{String}
+irrev_reaction_ids::Vector{String}
+metabolites::Vector{String}
+c::SparseVec
+S::SparseMat
+b::SparseVec
+xl::SparseVec
+xu::SparseVec
+C::SparseMat
+cl::Vector{Float64}
+cu::Vector{Float64}
+```
+"""
+mutable struct SMomentModel <: MetabolicModel
+    reaction_ids::Vector{String}
+    irrev_reaction_ids::Vector{String}
+    metabolites::Vector{String}
+    c::SparseVec
+    S::SparseMat
+    b::SparseVec
+    xl::SparseVec
+    xu::SparseVec
+end
+
+"""
+    stoichiometry(model::SMomentModel)
+
+Return stoichiometry matrix that includes enzymes as metabolites.
+"""
+stoichiometry(model::SMomentModel) = model.S
+
+"""
+    balance(model::SMomentModel)
+
+Return stoichiometric balance.
+"""
+balance(model::SMomentModel) = model.b
+
+"""
+    objective(model::SMomentModel)
+
+Return objective of `model`.
+"""
+objective(model::SMomentModel) = model.c
+
+"""
+    reactions(model::SMomentModel)
+
+Returns the reversible reactions in `model`. For 
+the irreversible reactions, use [`irreversible_reactions`][@ref].
+"""
+reactions(model::SMomentModel) = model.reaction_ids
+
+"""
+    n_reactions(model::SMomentModel)
+
+Returns the number of reactions in the model.
+"""
+n_reactions(model::SMomentModel) = length(model.reaction_ids)
+
+"""
+    irreversible_reactions(model::SMomentModel)
+
+Returns the irreversible reactions in `model`.
+"""
+irreversible_reactions(model::SMomentModel) = model.irrev_reaction_ids
```

Comment:
same as above
- `irreversible_reactions` → `reactions`
- `reactions` → `fluxes`
also, check that the matrix returned by `reaction_flux` is of the right size.